### DISCUSSION TITLE
How do I write LaTeX with colors in quarto presentations?

### DISCUSSION THREAD

**Discussion by uriahf** [quarto-dev/quarto-cli#4429] [Q&A] [answered]
I tried to do the same thing I used to do with Xaringan but it didn't work well:

```
---
title: "green and red"
format: revealjs
editor: visual
---

$$\begin{aligned}
\green{\bf{Net Benefit}} = \frac{\green{\bf{TP}}}{\text{N}} - \frac{\red{\bf{FP}}}{\text{N}} * {\frac{{p_{t}}}{{1 - p_{t}}}}
\end{aligned}$$
```

![image](https://user-images.githubusercontent.com/11351434/219983799-dd5befef-be04-40ae-86cf-db2effe409e3.png)

**Comment by baptiste** [comment, marked as answer]
Try this
```
$$\begin{aligned}
\color{green}{\bf{Net Benefit}} = \frac{\color{green}{\bf{TP}}}{\text{N}} - \frac{\color{red}{\bf{FP}}}{\text{N}} * {\frac{{p_{t}}}{{1 - p_{t}}}}
\end{aligned}$$
```

or, if you want to use those previous commands,

```
$$
\newcommand{\green}[1]{\color{green}{#1}}
\newcommand{\red}[1]{\color{red}{#1}}
$$

$$\begin{aligned}
\green{\bf{Net Benefit}} = \frac{\green{\bf{TP}}}{\text{N}} - \frac{\red{\bf{FP}}}{\text{N}} * {\frac{{p_{t}}}{{1 - p_{t}}}}
\end{aligned}$$
```

**Reply by cderv** [reply]
Just an additon for context. Math in HTML is supported through Mathjax, and color support is done with `\color` command.

**Comment by cderv** [comment]
> I tried to do the same thing I used to do with Xaringan

Can you share a **xaringan** example with this working ? 

I would like to compare how this would be working in **xaringan**. I have same issue on my side. 

About the issue itself, see previous answer from @baptiste which is the correct one. Math in HTML is using `Mathjax` and color is supported through the `\color` command. 

Thanks !

**Comment by uriahf** [comment]
Sure: [The code](https://github.com/uriahf/nb_step_by_step/blob/master/Decision_Curve_Step_by_Step.Rmd#:~:text=a%20Decision%20Curve%3F%3A-,.pull%2Dleft%5B,%5D,-.pull%2Dright%5B) and [the rendered page](https://nb-step-by-step.netlify.app/decision_curve_step_by_step#9).

By the way, I'm moving my slides from Xaringan to Quarto. 

Thank you so much for your answer, this community is just amazing!

**Reply by cderv** [reply]
Thanks. 

In your xaringan slide it was not by default. You added configuration for Mathjax 
```

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: {
    Macros: {
      red: ["{\\color{red}{#1}}", 1],
      green: ["{\\color{green}{#1}}", 1],
    },
    loader: {load: ['[tex]/color']},
    tex: {packages: {'[+]': ['color']}}
  }
});
</script>
```

You can probably do the same with Quarto. 

Or just do what we talked about before. 

I am reassured that I did not missed anything in xaringan 😅

Nice config by the way !

**Reply by RoyiAvital** [reply]
@cderv , Is there a way to load packages to MathJaX in Quarto?
Maybe have some options built in to Quarto to add packages (Like `color` above)?

**Reply by mcanouil** [reply]
MathJax is not the only Math JavaScript library.
it's only the default.
Adding such options would require to make it work with other libraries.
I don't think there are MathJax custom things currently in Quarto and I don't think that would be a good idea in terms of maintenance burden, in addition to making Quarto less agnostics of the Math library the users can use.

(That is only my opinion as a user)

**Reply by RoyiAvital** [reply]
@mcanouil , Let's apply the logic you presented on _Presentations_.
It means all features specific to RevealJS should be removed as it will require "maintenance burden" or having same features on other outputs.

Since this is not the case, it seems the policy is more related to the number of people using the feature.  
Like any product, features should reflect the needs of the users.

Since we have specific features for many cases, the questions is if this feature is as common as others.
I have no knowledge or data. My request is a single sample.

**Reply by cderv** [reply]
> Is there a way to load packages to MathJaX in Quarto?

@RoyiAvital did you try solution shared above ? You'll see it works with `\color` because the package is auloaded in Mathjax. 

If you read MathJaX doc, you understand that some packages are autoloaded by default, and other supported could be loaded using `\require`. 

Though, in Quarto, we use [`tex-chtml-full.js` ](https://docs.mathjax.org/en/latest/web/components/combined.html?highlight=tex-chtml-full.js#tex-chtml-full) version of MathJax. Looking at the link for the doc you'll see 
> The input/tex-full component loads the code for all the TeX extensions, and configures TeX to use all but the [physics](https://docs.mathjax.org/en/latest/input/tex/extensions/physics.html#tex-physics) and [colorv2](https://docs.mathjax.org/en/latest/input/tex/extensions/colorv2.html#tex-colorv2) extensions.

This means all available extensions are loaded except 2. But you can still load them using `\require`

Example: 

````
$$
\require{physics}
\dv{f}{x}
$$
````

Try this in a quarto doc with and without `\require{physics}`


Hope it helps understand how you can load support Mathjax TeX extension.

**Reply by RoyiAvital** [reply]
@cderv , This is perfect. I was really looking for the list of packages loaded by default.
I think it should go somewhere in the documentation.

I created a pull request for that: https://github.com/quarto-dev/quarto-web/pull/844.

Thanks.
